pkg/deploy/deploy.go

```diff
@@ -85,8 +97,133 @@ func (o *deployHandler) ApplyOpenShift(openshift v1alpha2.Component) error {
 // Execute will deploy the listed information in the `exec` section of devfile.yaml
 // We currently do NOT support this in `odo deploy`.
 func (o *deployHandler) Execute(command v1alpha2.Command) error {
-	// TODO:
-	// * Make sure we inject the "deploy" mode label once we implement exec in `odo deploy`
-	// * Make sure you inject the "component type" label once we implement exec.
-	return errors.New("exec command is not implemented for Deploy")
+	containerComps, err := generator.GetContainers(o.devfileObj, common.DevfileOptions{FilterByName: command.Exec.Component})
+	if err != nil {
+		return err
+	}
+	if len(containerComps) != 1 {
+		return fmt.Errorf("could not find the component")
+	}
+	containerComp := containerComps[0]
+
+	// Create a Kubernetes Job and use the container image referenced by command.Exec.Component
+	// Get the component for the command with command.Exec.Component
+	completionMode := batchv1.CompletionMode("Indexed")
+	job := batchv1.Job{
+		TypeMeta: generator.GetTypeMeta(kclient.JobsKind, kclient.JobsAPIVersion),
+		ObjectMeta: metav1.ObjectMeta{
+			// TODO: Check if the name is K8s valid.
+			Name: o.componentName + "-" + o.appName + "-" + command.Id + "-" + dfutil.GenerateRandomString(3), // TODO: Is there a function to return the standard odo names?
```

Comment:
TODO: Add length check on the name.

Reply:
1. Call GetContainers to get container for exec command
2. Change container fields to use exec commandLine
3. Use the container inside JobSpec.
4. JobSpec must be obtained from a generic function either inside libdevfile ~or the devfile library~.